File: solutions/scala/darts/1/src/main/scala/Darts.scala
object Darts:

    def score(x: Double, y: Double): Int =
        val distanceSquared = (x * x) + (y * y)
        distanceSquared match
            case _ if distanceSquared <= 1.0 => 10
            case _ if distanceSquared <= 25.0 => 5
            case _ if distanceSquared <= 100.0 => 1
            case _ => 0

File: solutions/kotlin/darts/1/src/main/kotlin/Darts.kt
object Darts {

	fun score(x: Number, y: Number): Int {
		val xCoordinate = x.toDouble()
		val yCoordinate = y.toDouble()

		val distanceSquared = xCoordinate * xCoordinate + yCoordinate * yCoordinate
		return when {
			distanceSquared <= 1
				-> 10

			distanceSquared <= 25
				-> 5

			distanceSquared <= 100
				-> 1

			else
				-> 0
		}
	}
}